lnbits/wallets/zbd.py

```diff
@@ -0,0 +1,171 @@
+            "expiresIn": 3600,
+            "callbackUrl": "",
+            "internalId": "",
+        }
+
+        r = await self.client.post(
+            "charges",
+            json=data,
+            timeout=40,
+        )
+
+        if r.is_error:
+            error_message = r.json()["message"]
+            return InvoiceResponse(False, None, None, error_message)
+
+        data = r.json()["data"]
+        checking_id = data["id"]  # this is a zbd id
+        payment_request = data["invoice"]["request"]
+        return InvoiceResponse(True, checking_id, payment_request, None)
+
+    async def pay_invoice(self, bolt11: str, fee_limit_msat: int) -> PaymentResponse:
+        # https://api.zebedee.io/v0/payments
+        r = await self.client.post(
+            "payments",
+            json={
+                "invoice": bolt11,
+                "description": "",
+                "amount": "",
+                "internalId": "",
+                "callbackUrl": "",
+            },
+            timeout=None,
+        )
+
+        if r.is_error:
+            error_message = r.json()["message"]
+            return PaymentResponse(False, None, None, None, error_message)
+
+        data = r.json()
+
+        # get the payment hash from the zbd api
+        decoded_request = await self.client.post(
+            "decode-invoice",
+            json={"invoice": bolt11},
+            timeout=40,
+        )
+        if decoded_request.is_error:
+            error_message = decoded_request.json()["message"]
+            return PaymentResponse(False, None, None, None, error_message)
+
+        decoded_data = decoded_request.json()
+
+        checking_id = decoded_data["data"]["paymentHash"]
+        fee_msat = -int(data["data"]["fee"])
+        preimage = data["data"]["preimage"]
+
+        return PaymentResponse(True, checking_id, fee_msat, preimage, None)
+
+    async def get_invoice_status(self, checking_id: str) -> PaymentStatus:
+        r = await self.client.get(f"charges/{checking_id}")
+        if r.is_error:
+            return PaymentStatus(None)
+        data = r.json()["data"]
+
+        statuses = {
+            "pending": None,
+            "paid": True,
+            "unpaid": None,
+            "expired": False,
+            "completed": True,
+        }
+        return PaymentStatus(statuses[data.get("status")])
+
+    async def get_payment_status(self, checking_id: str) -> PaymentStatus:
+        r = await self.client.get(f"payments/{checking_id}")
+        if r.is_error:
+            return PaymentStatus(None)
+
+        data = r.json()["data"]
+
+        statuses = {
+            "initial": None,
+            "pending": None,
+            "completed": True,
+            "error": None,
+            "expired": False,
+            "failed": False,
+        }
+
+        return PaymentStatus(statuses[data.get("status")], fee_msat=None, preimage=None)
+
+    async def paid_invoices_stream(self) -> AsyncGenerator[str, None]:
```

Comment:
If `ZBD webhook listener disabled` then the body of this method is not relevant.
Suggestion: just `log` and `return`

Reply:
ok